cmake/extensions.cmake

```diff
@@ -210,7 +210,14 @@ Please provide one of following: CONF_FILES")
     endif()
   endif()
 
-  zephyr_file(CONF_FILES ${NCS_FILE_CONF_FILES}/boards ${NCS_FILE_UNPARSED_ARGUMENTS})
+  set(additional_append)
+
+  if(DEFINED PREPROCESS_ARGS_BOARD)
+    parse_board_components(PREPROCESS_ARGS_BOARD board_name board_revision board_qualifiers)
+    set(additional_append BOARD ${board_name} BOARD_REVISION ${board_revision} BOARD_QUALIFIERS ${board_qualifiers})
+  endif()
```

Comment:
Also this code is bogus.
Doesn't matter if it works, but as mentioned elsewhere, then `ncs_file()` is an extension of `zephyr_file()`, and Zephyr file takes BOARD, BOARD_QUALIFIERS, and BOARD_REVISION as arbument, not `BOARD "plank@0.1.0/foo`.

Thus, code which calls `ncs_file()` should be corrected to pass required arguments instead of changing the function into accepting the board target.